aws-examples/mistral-e2e/chat.py

```diff
@@ -0,0 +1,62 @@
+# This is a simple command-line chat application that has contextual memory.
+
+from transformers import AutoTokenizer
+from optimum.neuron import NeuronModelForCausalLM
+
+model = NeuronModelForCausalLM.from_pretrained("./mistral_neuron", local_files_only=True)
+
+tokenizer = AutoTokenizer.from_pretrained("./mistral_neuron")
+tokenizer.pad_token_id = tokenizer.eos_token_id
+
+def format(message):
+    message = f"[INST] {message} [/INST]\n\n"
+    return message
+
+def format_chat_prompt(message, history, max_tokens):
+    chat = []
+
+    for interaction in history:
+        chat.append({"role": "user", "content": interaction[0]})
+        chat.append({"role": "assistant", "content": interaction[1]})
+
+    chat.append({'role': 'user', 'content': message})
+
+    for i in range(0, len(chat), 2):
+        prompt = tokenizer.apply_chat_template(chat[i:], tokenize=False)
+
+        tokens = tokenizer(prompt)
+        if len(tokens.input_ids) <= max_tokens:
+            return prompt
+
+    raise SystemError
```

Comment:
Why?

Reply:
Will add additional comments for explanation in upcoming commit.